indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -962,6 +992,32 @@ private void processMetricsSpec(final QueryableIndex index)
       }
     }
 
+    private void processMultiValuedColumns(final QueryableIndex index)
+    {
+      if (!needMultiValuedColumns) {
+        return;
+      }
+      // Process both dimensions and metrics since user-specified dimensions in compaction config may contain columns
+      // that were originally part of metrics.
+      for (String dimension : index.getAvailableDimensions()) {
+        if (isMultiValuedColumn(index, dimension)) {
+          multiValuedColumnsInfo.addMultiValuedColumn(dimension);
+        }
+      }
+      final AggregatorFactory[] aggregators = index.getMetadata().getAggregators();
+      if (aggregators != null) {
+        Arrays.stream(aggregators)
+              .filter(agg -> isMultiValuedColumn(index, agg.getName()))
+              .forEach(agg -> multiValuedColumnsInfo.addMultiValuedColumn(agg.getName()));
+      }
+    }
```

Comment:
Is this needed, shouldn't we be looking only at dimensions?

Reply:
Yes aggregates cannot be MVDs. Have removed them now.